### DISCUSSION TITLE
OAuth permissions for what necessary?

### DISCUSSION THREAD

**Discussion by passeee** [derekantrican/GAS-ICS-Sync#416] [General]
1. Allow this application to run when you are not present
2. Create, edit, organize, and delete all your tasks
3. Send email as you 
4. See, edit, share, and permanently delete all the calendars you can access using Google Calendar 
5. Connect to an external service 

1 & 4. makes sense. 
But why is 2. & 3. needed?
Also 5. does make sense but what external services? Like only outlook for the ICS or something else?

And where can I check the code where these permissions are necessary?

**Comment by passeee** [comment]
So I digged a bit into the code and managed to remove the permission for email, which is just a summary for events added/edited/removed
The Task permission could also be easily removed when needed
And the external service is once to fetch the ICS url and to fetch if there are new updates from github
So mandatory permissions are 1. & 4. & 5. and they look fine to me

**Comment by jonas0b1011001** [comment]
Correct, 1,4 and 5 are the bare minimum for automated syncing.

There is non way to request permissions 'only if needed' therefore the script will always require all permissions for all features, regardless if you use them or not (unless you are willing to change the code and remove them).
